packages/vscode-extension/src/devices/preview.ts

```diff
@@ -6,22 +6,69 @@ import { Logger } from "../Logger";
 import { Platform } from "../utilities/platform";
 import { RecordingData, TouchPoint } from "../common/Project";
 
-interface ReplayPromiseHandlers {
+type VideoRecordingPromiseType = "recording" | "replay";
+
+interface VideoRecordingPromiseHandlers {
   resolve: (value: RecordingData) => void;
   reject: (reason?: any) => void;
 }
 
 export class Preview implements Disposable {
+  private videoRecordingPromises: Map<
+    VideoRecordingPromiseType,
+    VideoRecordingPromiseHandlers | undefined
```

Comment:
I think we can just use `string` here as key, the keys that we use are not important here because we don't iterate over the map anywhere and we also don't have any special handling based on the key now.

We should not allow undefined value. Just remove entry from map when we want to reset it.